packages/renderer/src/lib/ui/NumberInput.svelte

```diff
@@ -30,7 +31,20 @@ function validateNumber() {
 }
 
 function onKeyPress(event: any) {
-  if (isNaN(Number(event.key))) {
+  // if type is integer it means we can only accept nu
+  // see https://json-schema.org/understanding-json-schema/reference/numeric
+  // Numbers with a zero fractional part are considered integers
+
+  // first, add the new character to the extisting value
+  const wantedValue = `${value}${event.key}`;
```

Comment:
The problem is that the test is done with the character added at the end, but the user can have placed the cursor anywhere.

For example for an integer input, if the input contains `12` and the cursor is between `1` and `2` and the user types `.`,  the character is accepted (because `12.` is considered as an integer), but the input contains `1.2` which is not an integer.

Reply:
fixed

Reply:
Nice fix . Don't forget to remove the debug logs

Reply:
fixed